beacon/beacon.go

```diff
@@ -577,41 +577,60 @@ func (pd *ProtocolDriver) initEpochStateIfNotPresent(logger log.Log, epoch types
 		// w1 is the weight units at δ before the end of the previous epoch, used to calculate `thresholdStrict`
 		// w2 is the weight units at the end of the previous epoch, used to calculate `threshold`
 		w1, w2 int
-		ontime = pd.clock.LayerToTime(epoch.FirstLayer())
+		ontime = pd.clock.LayerToTime(target.FirstLayer())
 		early  = ontime.Add(-1 * pd.config.GracePeriodDuration)
 	)
-	if err := pd.cdb.IterateEpochATXHeaders(epoch, func(header *types.ActivationTxHeader) error {
-		malicious, err := pd.cdb.IsMalicious(header.NodeID)
-		if err != nil {
-			return err
-		}
-		if !malicious {
-			epochWeight += header.GetWeight()
-		} else {
-			logger.With().Debug("malicious miner get 0 weight", log.Stringer("smesher", header.NodeID))
-		}
-		if _, ok := miners[header.NodeID]; !ok {
-			miners[header.NodeID] = &minerInfo{
-				atxid:     header.ID,
-				malicious: malicious,
+	query := `SELECT
+		atxs.id,
+		atxs.pubkey,
+		atxs.effective_num_units,
+		atxs.tick_count,
+		iif(i.proof is null, 0, 1) as malicious
+		FROM atxs left join identities i on atxs.pubkey = i.pubkey WHERE atxs.epoch = $1 `
+
+	_, err := pd.cdb.Exec(query,
```

Comment:
Same here, if possible we should use one of the existing functions in the `sql/atxs` package; otherwise we should move this into `sql/atxs`.